src/ClassMethodsHydrator.php

```diff
@@ -175,7 +175,7 @@ public function extract(object $object) : array
 
         $values = [];
 
-        if (null === $this->extractionMethodsCache[$objectClass]) {
+        if (empty($this->extractionMethodsCache[$objectClass])) {
```

Comment:
`empty()` should generally be avoided: besides that, can we perhaps add a test to verify the need for this change?

Reply:
Hello @Ocramius ,
> `empty()` should generally be avoided: besides that, can we perhaps add a test to verify the need for this change?

I could be more explicit: 

```php
if ([] === $this->extractionMethodsCache[$objectClass]) {
``` 
or
```php
if (! is_array($this->extractionMethodsCache[$objectClass]) || [] === $this->extractionMethodsCache[$objectClass]) {
```

since the only 2 direct assignments of this private array key are either to null or to an empty array. 

I am not sure about a test to prove that it is needed, since the returned value of the wrapping method is always an array (the array of extracted values, empty or not). I only noticed that the previous conditional code block makes sure it is always set/initialized to an array (if not already set or if reset in line 146) before reaching line 178.

May I ask why the `empty` construct should be avoided here (or elsewhere)?

kind regards

Reply:
We tend to avoid `empty()` as it is not specific enough. `empty()` evaluates to true for _anything_ false-y, which can lead to false positive results.